get/get.go

```diff
@@ -110,7 +168,70 @@ func (cmd *Cmd) list(ctx context.Context, client *api.Client, list runtimeclient
 		}
 	}
 
-	return nil
+	if cmd.AllProjects || !cmd.searchForName {
+		return nil
+	}
+
+	// we searched for a named object in the current project only, but did
+	// not find it there. Lets see if it was found in any other project.
+	if items.Len() == 0 {
+		// we did not find the named object in any other project, so
+		// we can return already
+		return nil
+	}
+	// we found the named object at least in one different project,
+	// so we return a hint to the user to search in these projects
+	var identifiedProjects []string
+	for i := 0; i < items.Len(); i++ {
+		// the "Items" field of a list type most often contains
+		// a slice of types, but not a slice of
+		// pointers to the type (e.g. "[]corev1.Pod" and not
+		// "[]*corev1.Pod"). We still want to check which
+		// variant (pointer or no pointer) implements the
+		// runtimeclient.Object interface.
+		obj, isRuntimeClientObj := getRuntimeObject(items.Index(i))
+		if !isRuntimeClientObj {
+			// very unlikely case: the items of the list did not
+			// implement runtimeclient.Object. As we can not get
+			// the project of the object, we just ignore it.
+			continue
+		}
+		identifiedProjects = append(identifiedProjects, obj.GetNamespace())
+	}
+	var result error
+	if len(identifiedProjects) > 0 {
+		h := &hint{
+			prefix: "NOTE",
+			msg: fmt.Sprintf(
+				"we found the resource in the following project(s): %s\n"+
+					"Maybe you want to use the '--project' "+
+					"option to specify one of these projects?\n",
+				strings.Join(identifiedProjects, " ,"),
+			),
+		}
```

Comment:
Can't we attach the hint to a custom error type (or just an error)? IMO the current get behavior is already wrong:

```
$ nctl get app doesnotexist
no Applications found in project nine
```

The exit code is 0 but should be 1 and it should not say "no applications found", it should say "application x not found in project y". And then with the new hint system, it could just be in the error string after that text.

Reply:
Sure, we can implement it like this. I was a bit hesitant to change already existing behavior, but I agree with your statements. `hint` is already a custom error, but would not be needed if we handle a 'your resource could not be found' error with exit code 1.

Reply:
I'm just thinking that if you need to explicitly handle the hint, we'll forget about it when we add new types. With the error you are forced to handle it.

Reply:
I adapted the code and implemented it as described above. Some examples:

```
$> ./nctl get app
PROJECT    NAME       HOSTS                               UNVERIFIED_HOSTS
nick       show-ip    show-ip.7b1b629.test.deploio.app    none

 $> ./nctl get app -A
PROJECT     NAME           HOSTS                                   UNVERIFIED_HOSTS
nick        show-ip        show-ip.7b1b629.test.deploio.app        none
nick-dev    dev-show-ip    dev-show-ip.d9d915b.test.deploio.app    none

 $> ./nctl get app show-ip
PROJECT    NAME       HOSTS                               UNVERIFIED_HOSTS
nick       show-ip    show-ip.7b1b629.test.deploio.app    none

 $> ./nctl get app dev-show-ip
nctl: error: resource "dev-show-ip" was not found in project nick, but it was found in project(s): nick-dev. Maybe you want to use the '--project' flag to specify one of these projects?

$> ./nctl get app dev-show-ip -A
PROJECT     NAME           HOSTS                                   UNVERIFIED_HOSTS
nick-dev    dev-show-ip    dev-show-ip.d9d915b.test.deploio.app    none

 $> ./nctl get mysql
no MySQLs found in project nick

$> ./nctl get mysql -A
no MySQLs found in any project
```